pandas/core/indexes/datetimelike.py

```diff
@@ -432,12 +433,36 @@ def asobject(self):
         return Index(self._box_values(self.asi8), name=self.name, dtype=object)
 
     def _convert_tolerance(self, tolerance):
-        try:
-            return Timedelta(tolerance).to_timedelta64()
-        except ValueError:
-            raise ValueError('tolerance argument for %s must be convertible '
-                             'to Timedelta: %r'
-                             % (type(self).__name__, tolerance))
+        tolerance = _list_to_ndarray(tolerance)
```

Comment:
just return
``np.asarray(pd.to_timedelta(tolerance))`` it will raise if it cannot convert

Reply:
you don't need any of the rest of the code

Reply:
here you *may* need to differentiate between a scalar an an array-like but not 100% sure